### PR TITLE
object_ladder

### PR DESCRIPTION
**Pull request by hack-nuss**
Object for Wooden Ladder

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Ladder/z_bg_ladder.c [hunk 1]
```diff
@@ -37,10 +37,10 @@ static InitChainEntry sInitChain[] = {
...
 
 static Gfx* sLadderDLists[] = {
-    object_ladder_DL_0000A0, // 12 Rung
-    object_ladder_DL_0002D0, // 16 Rung
-    object_ladder_DL_000500, // 20 Rung
-    object_ladder_DL_000730, // 24 Rung
+    gWoodenLadder12RungDL, // 12 Rung
+    gWoodenLadder16RungDL, // 16 Rung
+    gWoodenLadder20RungDL, // 20 Rung
+    gWoodenLadder24RungDL, // 24 Rung
```
Comment (engineer124) [inline src/overlays/actors/ovl_Bg_Ladder/z_bg_ladder.c:43]: Probably don't need these comments anymore

**Review by engineer124 — APPROVED** [review]

File: assets/xml/objects/object_ladder.xml [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
-        <DList Name="object_ladder_DL_0000A0" Offset="0xA0" /> <!-- Original name is "m2_HASIGO_45_model" -->
-        <Collision Name="object_ladder_Colheader_0001D8" Offset="0x1D8" /> <!-- Original name is "m2_HASIGO_45_bgdatainfo" -->
-        <DList Name="object_ladder_DL_0002D0" Offset="0x2D0" /> <!-- Original name is "m2_HASIGO_60_model" -->
-        <Collision Name="object_ladder_Colheader_000408" Offset="0x408" /> <!-- Original name is "m2_HASIGO_60_bgdatainfo" -->
-        <DList Name="object_ladder_DL_000500" Offset="0x500" /> <!-- Original name is "m2_HASIGO_75_model" -->
-        <Collision Name="object_ladder_Colheader_000638" Offset="0x638" /> <!-- Original name is "m2_HASIGO_75_bgdatainfo" -->
-        <DList Name="object_ladder_DL_000730" Offset="0x730" /> <!-- Original name is "m2_HASIGO_90_model" -->
-        <Collision Name="object_ladder_Colheader_000868" Offset="0x868" /> <!-- Original name is "m2_HASIGO_90_bgdatainfo" -->
-        <Texture Name="object_ladder_Tex_0008A0" OutName="tex_0008A0" Format="rgba16" Width="32" Height="32" Offset="0x8A0" />
+        <!-- Wooden Ladder with 12 Rung -->
```
Comment (tom-overton) [inline assets/xml/objects/object_ladder.xml:4]: This is me just being unbelievably pedantic, sorry:
```suggestion
        <!-- Wooden Ladder with 12 Rungs -->
```

File: assets/xml/objects/object_ladder.xml [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+        <DList Name="gWoodenLadder12RungDL" Offset="0xA0" /> <!-- Original name is "m2_HASIGO_45_model" -->
+        <Collision Name="gWoodenLadder12RungCol" Offset="0x1D8" /> <!-- Original name is "m2_HASIGO_45_bgdatainfo" -->
+
+        <!-- Wooden Ladders with 16 Rung -->
```
Comment (tom-overton) [inline assets/xml/objects/object_ladder.xml:8]:
```suggestion
        <!-- Wooden Ladder with 16 Rungs -->
```

File: assets/xml/objects/object_ladder.xml [hunk 4]
```diff
@@ -1,13 +1,23 @@
...
+        <DList Name="gWoodenLadder16RungDL" Offset="0x2D0" /> <!-- Original name is "m2_HASIGO_60_model" -->
+        <Collision Name="gWoodenLadder16RungCol" Offset="0x408" /> <!-- Original name is "m2_HASIGO_60_bgdatainfo" -->
+
+        <!-- Wooden Ladder with 20 Rung -->
```
Comment (tom-overton) [inline assets/xml/objects/object_ladder.xml:12]:
```suggestion
        <!-- Wooden Ladder with 20 Rungs -->
```

File: assets/xml/objects/object_ladder.xml [hunk 5]
```diff
@@ -1,13 +1,23 @@
...
+        <DList Name="gWoodenLadder20RungDL" Offset="0x500" /> <!-- Original name is "m2_HASIGO_75_model" -->
+        <Collision Name="gWoodenLadder20RungCol" Offset="0x638" /> <!-- Original name is "m2_HASIGO_75_bgdatainfo" -->
+
+        <!-- Wooden Ladder with 24 Rung -->
```
Comment (tom-overton) [inline assets/xml/objects/object_ladder.xml:16]:
```suggestion
        <!-- Wooden Ladder with 24 Rungs -->
```

**Review by AngheloAlf — APPROVED** [review]

**Review by tom-overton — APPROVED** [review]

**Pull request merged**